integration-tests/oidc-wiremock/src/test/java/io/quarkus/it/keycloak/KeycloakTestResource.java

```diff
@@ -71,8 +75,37 @@
         // Invalid
         defineInvalidIntrospectionMockTokenStubForUserWithRoles("expired", emptySet());
 
+        // Login Page
+        server.stubFor(
+                get(urlPathMatching("/auth/realms/quarkus"))
+                        .withQueryParam("redirect_uri", equalTo("http://localhost:8081/code-flow"))
+                        .willReturn(aResponse()
+                                .withHeader("Content-Type", MediaType.TEXT_HTML)
+                                .withBody("<html>\n" +
+                                        "<body>\n" +
+                                        " <form action=\"/login\" name=\"form\">\n" +
+                                        "  <input type=\"text\" id=\"username\" name=\"username\"/>\n" +
+                                        "  <input type=\"password\" id=\"password\" name=\"password\"/>\n" +
+                                        "  <input type=\"hidden\" id=\"state\" name=\"state\" value=\"SOME_STATE\"/>\n" +
+                                        "  <input type=\"hidden\" id=\"redirect_uri\" name=\"redirect_uri\" value=\"SOME_REDIRECT\"/>\n"
+                                        +
+                                        "  <input type=\"submit\" id=\"login\" value=\"login\"/>\n" +
+                                        "</form>\n" +
+                                        "</body>\n" +
+                                        "</html> ")));
+
+        // Login Request
+        server.stubFor(
```

Comment:
It won't work - these `state` and `redirect_uri` will have to be captured from your form, and you should redirect to this redirect_uri. Also, username and password will likely be coming in the form payload

Reply:
Hello @sberyozkin ,

Yes I was kinda messing around trying to figure things out 😄 

Now after the login button is clicked via HtmlUnit the following request is being called

`http://localhost:54913/login?username=alice&password=alice&state=d25a2924-5da4-45dd-8a0a-c7758e375f36&redirect_uri=http%3A%2F%2Flocalhost%3A8081%2Fcode-flow`

I currently mock this request as a request that starts with `/login` and has a query parameter such as `redirect_uri=http://localhost:8081/code-flow` which returns only a http `200` OK. What should we be returning as a reponse?

Reply:
@cemnura Sure, here I believe you should also use the templating featiure to capture the `redirect_uri` (as opposed to hardcoding it) and also `state` - now both will be returned after HtmlUnit `login` is pressed. Status should be `302`.

It is nearly there :-), thanks

Reply:
Thanks for your patience and help :) 

Yes we are very close I think I got it 😄

Reply:
Thanks for your patience and help :) 

Yes we are very close I think I got it 😄  

I got the final response to return the following

```
{
  "status" : 302,
  "headers" : {
    "Location" : "http://localhost:8081/index.html",
    "state" : "e7a8b02c-1c37-4d1b-932e-3bfda399735e"
  },
  "transformers" : [ "response-template" ]
}
```

Reply:
@cemnura Great stuff, thanks for persevering, it is appreciated :-).
I think we still need to fix something here, I'm not sure what, because the `state` must be returning as a query parameter, I don't understand yet why the test passes, I'll have a look and let you know :-)

Thanks

Reply:
@cemnura I see where the problem is, `index.html` is a public resource as we have not configured it as a protected one, something like
```
quarkus.http.auth.permission.roles1.paths=/index.html
quarkus.http.auth.permission.roles1.policy=authenticated
```

but we also have a no-op `CodeFlowResource`.

I propose to simplify as follows:
- remove `redirect-path` configuration (which now points to `index.html`), Quarkus will use the current request URI by default.
- remove `META-INF/resources/index.html`
- update `CodeFlowResource` to have `SecurityIdentity` injected and have it is current empty method return the principal name
- update the test to expect a response from `CodeFlowResource`

Now you should see your test failing :-). Returning `state` as a query parameter should fix it :-)

Give it a try please

Reply:
@cemnura, oh, I forgot, I typed it in the linked issue or somewhere above, but you also need to return a `code` query parameter, alongside the saved `state` one. `code` value can be hardcoded and then you also need to stub another token endpoint request which would look like this
```
grant_type=authorization_code&code=some_code&redirect_uri=the_same_redirect_uri
```

It implies that we indeed need to also hardcode `redirect_uri` - so you can simplify the templating code a bit and avoid saving/capturing `redirect_uri` and have it hardcoded as pointing to the `CodeFlowResource` method, as you had it earlier

That should be really it :-)